File: src/main/java/me/kuku/yuq/service/impl/QQJobServiceImpl.java
package me.kuku.yuq.service.impl;

import com.icecreamqaq.yudb.jpa.annotation.Transactional;
import me.kuku.yuq.dao.QQJobDao;
import me.kuku.yuq.entity.QQJobEntity;
import me.kuku.yuq.service.QQJobService;

import javax.inject.Inject;
import java.util.List;

public class QQJobServiceImpl implements QQJobService {
    @Inject
    private QQJobDao qqJobDao;
    @Override
    @Transactional
    public QQJobEntity findByQQAndType(Long qq, String type) {
        return qqJobDao.findByQQAndType(qq, type);
    }

    @Override
    @Transactional
    public List<QQJobEntity> findByQQ(Long qq) {
        return qqJobDao.findByQQ(qq);
    }

    @Override
    @Transactional
    public List<QQJobEntity> findByType(String type) {
        return qqJobDao.findByType(type);
    }

    @Override
    @Transactional
    public void delByQQ(Long qq) {
        qqJobDao.delByQQ(qq);
    }

    @Override
    @Transactional
    public void save(QQJobEntity qqJobEntity) {
        qqJobDao.saveOrUpdate(qqJobEntity);
    }
}

File: src/main/kotlin/me/kuku/yuq/service/impl/QQJobServiceImpl.kt
package me.kuku.yuq.service.impl

import com.icecreamqaq.yudb.jpa.annotation.Transactional
import me.kuku.yuq.dao.QQJobDao
import me.kuku.yuq.entity.QQJobEntity
import me.kuku.yuq.service.QQJobService
import javax.inject.Inject

class QQJobServiceImpl: QQJobService {

    @Inject
    private lateinit var qqJobDao: QQJobDao

    @Transactional
    override fun findByQQAndType(qq: Long, type: String) = qqJobDao.findByQQAndType(qq, type)

    @Transactional
    override fun findByQQ(qq: Long) = qqJobDao.findByQQ(qq)

    @Transactional
    override fun findByType(type: String) = qqJobDao.findByType(type)

    @Transactional
    override fun delByQQ(qq: Long) = qqJobDao.delByQQ(qq)

    @Transactional
    override fun save(qqJobEntity: QQJobEntity) = qqJobDao.saveOrUpdate(qqJobEntity)
}